mempool_updatefromblock.py fails on 32-bit (armhf, ...)

Steps to reproduce:

* armhf metal, or via `podman run -it --rm --platform linux/arm/v7 'debian:trixie'`
* `export DEBIAN_FRONTEND=noninteractive && apt update && apt install curl wget htop git vim ccache -y && git clone https://github.com/bitcoin/bitcoin.git  --depth=1 ./b-c && cd b-c && apt install build-essential cmake pkg-config  python3-zmq libzmq3-dev libevent-dev libboost-dev libsqlite3-dev  systemtap-sdt-dev  libcapnp-dev capnproto  libqrencode-dev qt6-tools-dev qt6-l10n-tools qt6-base-dev  clang llvm libc++-dev libc++abi-dev   -y`
* `rm -rf ./bld-bisect/ && cmake -B ./bld-bisect -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER='clang' -DCMAKE_CXX_COMPILER='clang++'  --preset=dev-mode -DBUILD_UTIL_CHAINSTATE=OFF -DBUILD_BENCH=OFF -DBUILD_FUZZ_BINARY=OFF -DBUILD_GUI=OFF -DBUILD_KERNEL_LIB=OFF -DBUILD_TESTS=OFF -DENABLE_WALLET=OFF -DENABLE_IPC=OFF -DWITH_ZMQ=OFF -DWITH_BDB=OFF -DWITH_NATPMP=OFF -DWITH_NATPMPP=OFF -DWITH_MINIUPNPC=OFF -DWITH_MULTIPROCESS=OFF`
* `cmake --build ./bld-bisect --parallel  $(nproc)  &&  ./bld-bisect/test/functional/mempool_updatefromblock.py --timeout-factor 5`

First bad commit is 47894367b583c06c53d568dc4a984d27bac8f742


```
2025-12-18T20:39:45.629000Z TestFramework (INFO): PRNG seed is: 1933782243
2025-12-18T20:39:45.630000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_b7nlv98l
2025-12-18T20:39:47.600000Z TestFramework (INFO): Creating 100 transactions...
2025-12-18T20:39:50.975000Z TestFramework (INFO): The batch of 25 transactions has been accepted into the mempool.
2025-12-18T20:39:51.183000Z TestFramework (INFO): All of the transactions from the current batch have been mined into a block.
2025-12-18T20:39:59.928000Z TestFramework (INFO): The batch of 25 transactions has been accepted into the mempool.
2025-12-18T20:40:00.151000Z TestFramework (INFO): All of the transactions from the current batch have been mined into a block.
2025-12-18T20:40:14.350000Z TestFramework (INFO): The batch of 25 transactions has been accepted into the mempool.
2025-12-18T20:40:14.574000Z TestFramework (INFO): All of the transactions from the current batch have been mined into a block.
2025-12-18T20:40:30.226000Z TestFramework (INFO): The last batch of 25 transactions has been accepted into the mempool.
2025-12-18T20:40:32.193000Z TestFramework (INFO): All of the recently mined transactions have been re-added into the mempool in 1.9641704559326172 seconds.
2025-12-18T20:40:32.194000Z TestFramework (INFO): Checking descendants/ancestors properties of all of the in-mempool transactions...
2025-12-18T20:40:34.347000Z TestFramework (INFO): Creating independent transactions to test MAX_DISCONNECTED_TX_POOL_BYTES limit during reorg
2025-12-18T20:42:25.872000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/b-c/test/functional/test_framework/test_framework.py", line 183, in main
    self.run_test()
    ~~~~~~~~~~~~~^^
  File "/b-c/./bld-bisect/test/functional/mempool_updatefromblock.py", line 185, in run_test
    self.test_max_disconnect_pool_bytes()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/b-c/./bld-bisect/test/functional/mempool_updatefromblock.py", line 175, in test_max_disconnect_pool_bytes
    assert_equal(len(mempool), expected_parent_count * 2)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/b-c/test/functional/test_framework/util.py", line 77, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(398 == 396)

--- a/ci/test/00_setup_env_arm.sh
+++ b/ci/test/00_setup_env_arm.sh
@@ -12,8 +12,6 @@ export PACKAGES="python3-zmq g++-arm-linux-gnueabihf libc6:armhf libstdc++6:armh
 export CONTAINER_NAME=ci_arm_linux
 export CI_IMAGE_NAME_TAG="mirror.gcr.io/ubuntu:24.04"  # Check that https://packages.ubuntu.com/noble/g++-arm-linux-gnueabihf (version 13.x, similar to guix) can cross-compile
 export CI_IMAGE_PLATFORM="linux/arm64"
-export RUN_UNIT_TESTS=true
-export RUN_FUNCTIONAL_TESTS=false
 export GOAL="install"
 export CI_LIMIT_STACK_SIZE=1
 # -Wno-psabi is to disable ABI warnings: "note: parameter passing for argument of type ... changed in GCC 7.1"

--- a/test/functional/mempool_updatefromblock.py
+++ b/test/functional/mempool_updatefromblock.py
@@ -13,7 +13,7 @@ import time
 
 from test_framework.blocktools import create_empty_fork
 from test_framework.test_framework import BitcoinTestFramework
-from test_framework.util import assert_equal, assert_raises_rpc_error
+from test_framework.util import assert_equal, assert_greater_than_or_equal, assert_raises_rpc_error
 from test_framework.wallet import MiniWallet
 from test_framework.mempool_util import DEFAULT_CLUSTER_LIMIT
 
@@ -161,12 +161,17 @@ class MempoolUpdateFromBlockTest(BitcoinTestFramework):
         # but not all, and any time parent is dropped, child is also removed
         self.trigger_reorg(fork_blocks=fork_blocks)
         mempool = self.nodes[0].getrawmempool()
-        expected_parent_count = len(large_std_txs) - 2
-        assert_equal(len(mempool), expected_parent_count * 2)
+        # At least one parent must be dropped, but more may be dropped,
+        # depending on the dynamic cost overhead.
+        expected_parent_count = len(large_std_txs) - 1
+        assert_greater_than_or_equal(expected_parent_count * 2, len(mempool))
+        expected_parent_count = len(mempool) // 2
+
+        parent_presence = [tx["txid"] in mempool for tx in large_std_txs]
 
         # The txns at the end of the list, or most recently confirmed, should have been trimmed
-        assert_equal([tx["txid"] in mempool for tx in large_std_txs], [tx["txid"] in mempool for tx in small_child_txs])
-        assert_equal([tx["txid"] in mempool for tx in large_std_txs], [True] * expected_parent_count + [False] * 2)
+        assert_equal(parent_presence, [tx["txid"] in mempool for tx in small_child_txs])
+        assert_equal(parent_presence, [True] * expected_parent_count + [False] * (len(large_std_txs) - expected_parent_count))
 
     def test_chainlimits_exceeded(self):
         self.log.info('Check that too long chains on reorg are handled')

